### DISCUSSION TITLE
Getting a specific chain SAS

### DISCUSSION THREAD

**Discussion by Marwa-Da** [molstar/molstar#909] [Q&A] [answered]
Hi all,

I have the surface areas (SAS) for all residues based on the AccessibleSurfaceArea behavior. I know that that the chain's area is the sum of its residues' areas. How to get all chains' areas programmatically?

P.S. In fact I don't know how to get the chains in general, either by atoms or by residue. Any explanation related to this is really appreciated as my knowledge regarding getting the sequences, chains,.. etc is low.

**Comment by Marwa-Da** [comment]
@dsehnal Can you lend me a helping hand please?

**Reply by dsehnal** [reply, marked as answer]
Maybe you can try the `Structure.eachAtomicElement` (or perhaps the name is a bit different) and in the residue callback you can transform the provided location to a chain identifier (asym id) using `StructureProperties.chain`.

**Comment by Marwa-Da** [comment]
For other developers coming after me, this is what helped me do the task:
`    getAsymId(unit) {
        switch (unit.kind) {
            case Unit.Kinds.Atomic:
                return StructureProperties.chain.label_asym_id;
            case Unit.Kinds.Spheres:
            case Unit.Kinds.Gaussians:
                return StructureProperties.coarse.asym_id;
        }
    }`
and this:
`Structure.eachAtomicHierarchyElement`

**Reply by dsehnal** [reply]
The `Structure.eachAtomicHierarchyElement` only iterates over atomic units and the `structure properties` do the cast automatically, so you only need `StructureProperties.chain.label_asym_id(loc)` to get the asym id.

**Reply by Marwa-Da** [reply]
Ok, thank you for the correction! You're right, it's obvious from the function name "each**Atomic**Hierarchy**Element**"
